app/src/main/java/org/oppia/app/player/audio/AudioViewModel.kt

```diff
@@ -0,0 +1,50 @@
+package org.oppia.app.player.audio
+
+import androidx.fragment.app.FragmentManager
+import androidx.lifecycle.ViewModel
+import org.oppia.app.fragment.FragmentScope
+import javax.inject.Inject
+
+private const val TAG_DIALOG = "LANGUAGE_DIALOG"
+
+/** [ViewModel] for user app usage history. */
+@FragmentScope
+class AudioViewModel @Inject constructor(
+) : ViewModel() {
+
+  var currentlanguageCode: String = "en"
+
+  val audioLanguageCode: String? by lazy {
+    getCurrentLanguageCode()
+  }
+
+  fun setAudioLanguageCode(languageCode: String) {
+    currentlanguageCode = languageCode
+  }
+
+  private fun getCurrentLanguageCode(): String {
+    return currentlanguageCode
+  }
+
+  fun languageSelectionClicked(fragmentManager: FragmentManager) {
+    val fragmentTransaction = fragmentManager.beginTransaction()
+    val prev = fragmentManager.findFragmentByTag(TAG_DIALOG)
+    if (prev != null) {
+      fragmentTransaction.remove(prev)
+    }
+    fragmentTransaction.addToBackStack(null)
+    val dialogFragment = LanguageDialogFragment.newInstance(
+      getDummyAudioLanguageList(),
+      "en"
+    )
+    dialogFragment.show(fragmentManager, TAG_DIALOG)
```

Comment:
Prefer ``showNow()`` to synchronously show it. The async commit()/show() methods can result in some tricky bugs that are difficult to track down & fix without committing with state loss, and that leads to its own share of issues.

Reply:
Was not aware of `showNow()` Thanks for this point.